plugins/SitesManager/Controller.php

```diff
@@ -24,12 +24,23 @@
 use Piwik\View;
 use Piwik\Http;
 use Piwik\Plugins\SitesManager\GtmSiteTypeGuesser;
+use Matomo\Cache\Lazy;
+use Piwik\Cache;
 
 /**
  *
  */
 class Controller extends \Piwik\Plugin\ControllerAdmin
 {
+    /** @var Lazy */
+    private $cache;
+
+    public function __construct() {
+        $this->cache = Cache::getLazyCache();
+
+        parent::__construct();
+    }
```

Comment:
Is this constructor necessary? Would it work to inject it like:

```
/** @var Lazy */
private $cache;

public function __construct(Lazy $cache)
{
    $this->cache = $cache;
    parent::__construct();
}
```
?

Reply:
yes, you're right, it works